libnmstate/nm/connection.py

```diff
@@ -117,7 +117,7 @@ def delete(self):
     def activate(self):
         if self.con_id:
             self.import_by_id()
-        elif self.nmdevice:
+        elif not self.profile and self.nmdevice:
```

Comment:
This code change does not match the commit message. If `self.import_by_id()` fails, `self.import_by_device()` would still not be called. Also `not self.profile` is used in two conditions, so if this code is correct, then something like

```
elif not self.profile:
    if self.nmdevice:
        ....
    else:
        raise(...)
```

Reply:
Sorry, by "fails" it means `import_by_id` has not found any profile with that id. Therefore, we should try `import_by_device` if `self.nmdevice` is not None. If after trying `import_by_device` the profile still being None, then nmstate will raise.

IMO, the current logic is correct. I could rephrase the commit message.

Reply:
Is it an error if `self.profile` is not None? The commit will make it an error while it wasn't before.

Reply:
> Sorry, by "fails" it means `import_by_id` has not found any profile with that id. Therefore, we should try `import_by_device` if `self.nmdevice` is not None. If after trying `import_by_device` the profile still being None, then nmstate will raise.


The code calls `self.import_by_id()` or `self.import_by_device()` The logic does not make it call `self.import_by_device()` based on the result of `self.import_by_id()` unless I am missing something here.

Reply:
Please, see the new approach.